WICKET-7172: Extend CSP with support for script-src-attr, style-src-attr

keesvandieren opened the pull request:
Those new directives have been added to CSP in 2022 but are not yet available in Wicket

[WICKET-7172](https://issues.apache.org/jira/browse/WICKET-7172)

Commit:
Add support for script-src-attr, style-src-attr

martin-g reviewed (approved):
We need a JIRA ticket for the changelog.

keesvandieren commented:
@martin-g created https://issues.apache.org/jira/browse/WICKET-7172

bitstorm committed:
Fixed brackets indentation

bitstorm committed:
Fixed brackets indentation

bitstorm committed:
refactor for logic condition

Commit:
Drop unsupported directive

Not listed here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Security-Policy

martin-g committed:
WICKET-7172: Improve the unit tests

martin-g commented on wicket-core/src/main/java/org/apache/wicket/csp/CSPDirective.java:
```diff
@@ -147,6 +151,19 @@ public String getValue()
+				throw new IllegalArgumentException("Directive " + this + " supports only one value");
+			}
+
+			if (value != CSPDirectiveSrcValue.NONE && value != CSPDirectiveSrcValue.UNSAFE_INLINE) 
```
```suggestion
			if (!CSPDirectiveSrcValue.NONE.equals(value) && !CSPDirectiveSrcValue.UNSAFE_INLINE.equals(value)) 
```

martin-g committed:
Use equals() to compare the values

keesvandieren commented on wicket-core/src/main/java/org/apache/wicket/csp/CSPDirective.java:
```diff
@@ -147,6 +151,19 @@ public String getValue()
+		{
+			if (!existingDirectiveValues.isEmpty()) 
+			{
+				throw new IllegalArgumentException("Directive " + this + " supports only one value");
```
Technically this is true, but in practice only 2 values are allowed: none and 'unsafe inline'. They contradict each other so shouldn't be used together.

keesvandieren committed:
Merge branch 'apache:master' into add-src-attrs-to-csp-directive

Commit:
Correct implementation for script-src-attributes and style-src-attributes

Support additional values from spec

Removed specific verifications

keesvandieren commented:
@martin-g  / @bitstorm  I have corrected the implementation. 

Can you please verify the PR? Thanks in advance!

bitstorm reviewed (approved):
I also don't have a huge experience with CSP directives but it looks good
